app/controllers/users_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  include ControllerHelper
+  before_action :user, only: %i[show edit update destroy]
+
+  def index
+    @users = User.page params[:page]
+  end
+
+  def show; end
+
+  def edit; end
+
+  def destroy
+    @user.destroy
+    redirect_to [:users]
+    flash[:danger] = 'User profile has been deleted'
+  end
+
+  def update
+    if @user.update(user_params)
+      successful_update("User profile \"#{@user.full_name}\"  updated")
+      redirect_to @user
+    else
+      flash.now[:warning] = 'Invalid parameters for editing!'
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:first_name, :last_name, :email, :avatar, :role, :password)
```

Comment:
seems like you don't have `role` and `password` fields in the form. it means those fields are not expected to be met here, so remove them from the permitted list

Reply:
done